### PR TITLE
Add detailed contributing documentation

### PR DESCRIPTION
**Pull request by alastair**
An attempt at starting detailed contributing documentation so that we don't have to keep repeating ourselves. Based a little bit on discussion in IRC, on the Forums, and things that I've noticed over the last few years.

My preference is to get this merged as quickly as possible, and iterate on changes as issues come up; rather than spending a long time discussing updates in this PR.
I'm open to corrections of things which are obviously wrong and updates of things that we currently do.

We should also consider if this is a good place to have these guidelines. CB has a nice set of documentation online (http://critiquebrainz.readthedocs.io/contributing.html). We could also replicate this.

### CODE REVIEW

**Comment by gentlecat** [comment]
It would be nice if you could move some of the common stuff to https://github.com/metabrainz/guidelines and put a link to that page. And then expand on AcousticBrainz-specific suggestions/requirements.

**Comment by alastair** [comment]
:+1: I didn't know about this repo

**Comment by alastair** [comment]
pointed to metabrainz guidelines!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,18 +1,61 @@
-## Contributing
+# AcousticBrainz contributing guidelines
```
Comment (gentlecat) [inline CONTRIBUTING.md]: Would `contribution guidelines` be more correct? Otherwise to me it sounds like guidelines are contributing to something. :)

**Pull request merged**